symengine/test_visitors.h

```diff
@@ -9,35 +9,35 @@ namespace SymEngine
 class ZeroVisitor : public BaseVisitor<ZeroVisitor>
 {
 private:
-    bool zero_; // true = testing for zero, false = testing for nonzero
     tribool is_zero_;
-    bool neither_ = false; // Neither zero nor non-zero, i.e. not a number
+    const Assumptions *assumptions_;
 
-public:
-    ZeroVisitor(bool zero) : zero_{zero} {}
-    void bvisit(const Basic &x)
+    void error()
     {
-        is_zero_ = tribool::indeterminate;
+        throw SymEngineException(
+            "Only numeric types allowed for is_zero/is_nonzero");
     };
-    void bvisit(const Symbol &x)
+
+public:
+    ZeroVisitor(const Assumptions *assumptions) : assumptions_(assumptions) {}
+
+    void bvisit(const Basic &x)
     {
         is_zero_ = tribool::indeterminate;
     };
+    void bvisit(const Symbol &x);
     void bvisit(const Number &x);
     void bvisit(const Set &x)
     {
-        is_zero_ = tribool::trifalse;
-        neither_ = true;
+        error();
```

Comment:
nevermind - class member functions defined in a header are implicitly inline

Reply:
Not in C++ 20 modules so using explicit `inline` or putting them in the cpp is good style I think. I will change if not just to see if codecov has a beef with this.

Reply:
I moved the inlined member functions to the cpp now. `inline` could  for small functions help the compiler create more performant code, but I don't think so for double dispatch visitors since the member functions are never dispatched directly. This way compile time will be reduced. I am curious to see if codecov likes this better, but we won't know until the github actions outage is fixed.

Reply:
Now all checks pass. I'll keep it this way.